core/sql/executor/ex_sort.cpp

```diff
@@ -330,21 +364,39 @@ ExSortTcb::~ExSortTcb()
    nfDiags_ = NULL;
   
   freeResources();
-  
-  if(pool_)
-    delete pool_;
+
 };
   
 ////////////////////////////////////////////////////////////////////////
 // Free Resources
 //
 void ExSortTcb::freeResources()
 {
-  if (sortPool_)
+  if (partialSortPool_)
+  {
+    delete partialSortPool_;
+    partialSortPool_ = NULL;
+  }
+  if (regularSortPool_)
+  {
+    delete regularSortPool_;
+    regularSortPool_ = NULL;
+  }
+  if (topNSortPool_)
+  {
+    delete topNSortPool_;
+    topNSortPool_ = NULL;
+  }
+  if (sortPool_ && (sortPool_ != receivePool_))
```

Comment:
I will make this change. The thinking was to delete the memory once, either by sortPool_ or receivePool_.  setting them to null seems to be a no op since it is called only in the destructor.